lib/api/apiUtils/rateLimit/refillJob.js

```diff
@@ -0,0 +1,130 @@
+const werelogs = require('werelogs');
+
+const { getAllTokenBuckets, cleanupTokenBuckets } = require('./tokenBucket');
+
+const logger = new werelogs.Logger('S3');
+
+let refillInterval = null;
+
+// Refill interval in milliseconds (how often to check and refill buckets)
+const REFILL_INTERVAL_MS = 100;
+
+// Cleanup interval for expired buckets (every 10 seconds)
+const CLEANUP_INTERVAL_MS = 10000;
+let cleanupCounter = 0;
+
+/**
+ * Background refill job for token buckets
+ *
+ * This job runs periodically (default: every 100ms) and:
+ * 1. Iterates through all active token buckets
+ * 2. Triggers async refill for buckets below threshold
+ * 3. Periodically cleans up expired/idle token buckets
+ *
+ * The refills are asynchronous and non-blocking, keeping Redis
+ * out of the hot request path.
+ */
+async function refillTokenBuckets() {
+    const tokenBuckets = getAllTokenBuckets();
+
+    if (tokenBuckets.size === 0) {
+        return {
+            checked: 0,
+            refilled: 0,
+        };
+    }
```

Comment:
It may not be needed, unless it's an optimization to avoid calling `Promise.all` (not sure it's worth it, but fine with it)